dart/lib/src/hub.dart

```diff
@@ -0,0 +1,280 @@
+    _isEnabled = true;
+  }
+
+  static void _validateOptions(SentryOptions options) {
+    if (options == null) {
+      throw ArgumentError.notNull('options');
+    }
+
+    if (options.dsn == null) {
+      throw ArgumentError.notNull('options.dsn');
+    }
+  }
+
+  bool _isEnabled = false;
+
+  /// Check if the Hub is enabled/active.
+  bool get isEnabled => _isEnabled;
+
+  SentryId _lastEventId;
+
+  /// Last event id recorded in the current scope
+  SentryId get lastEventId => _lastEventId;
+
+  /// Captures the event.
+  Future<SentryId> captureEvent(Event event) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureEvent' call is a no-op.",
+      );
+    } else if (event == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureEvent called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureEvent(event, scope: item.scope);
+        } catch (err) {
+          /* TODO add Event.id */
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing event with id: ${event}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureEvent',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  /// Captures the exception
+  Future<SentryId> captureException(
+    dynamic throwable, {
+    dynamic stackTrace,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureException' call is a no-op.",
+      );
+    } else if (throwable == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureException called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          // TODO pass the scope
+          sentryId = await item.client.captureException(
+            throwable,
+            stackTrace: stackTrace,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing exception : ${throwable}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureException',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  /// Captures the message.
+  Future<SentryId> captureMessage(
+    Message message, {
+    SeverityLevel level = SeverityLevel.info,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
```

Comment:
Seems we log and leave the if block, assign the sentryId into the _lastId bit but no event was captured which is misleading.

Maybe we just `return` here?

Reply:
the Java SDK does pretty much the same, the `lastEventId` will be empty if an event failed (eg SDK disabled) to be captured for some reason, I will check other SDKs

Reply:
https://github.com/getsentry/sentry-dart/pull/113#discussion_r507580774